gbpservice/nfp/base_mode_vm/api/v1/controllers/controller.py

```diff
@@ -167,3 +159,65 @@ def _format_description(self, msg):
 
         error_data = {'failure_desc': {'msg': msg}}
         return error_data
+
+
+    def _add_routes(self, route_info):
+        source_cidrs = route_info['resource_data']['source_cidrs']
+        consumer_cidr = route_info['resource_data']['destination_cidr']
+        gateway_ip = route_info['resource_data']['gateway_ip']
+        for cidr in source_cidrs:
+            source_interface = self._get_if_name_by_cidr(cidr)
+            try:
+                interface_number_string = source_interface.split("eth",1)[1]
+            except IndexError:
+                LOG.error("Retrieved wrong interface %s for configuring "
+                             "routes" %(source_interface))
+            routing_table_number = 20 + int(interface_number_string)
+            ip_rule_command = "ip rule add from %s table %s" %(
+                cidr, routing_table_number)
+            out1 = subprocess.Popen(ip_rule_command, shell=True,
+                                    stdout=subprocess.PIPE).stdout.read()
+            ip_rule_command = "ip rule add to %s table main" %(cidr)
+            out2 = subprocess.Popen(ip_rule_command, shell=True,
+                                    stdout=subprocess.PIPE).stdout.read()
+            ip_route_command = "ip route add table %s default via %s" %(
+                                    routing_table_number, gateway_ip)
```

Comment:
Keep all of the route commands in try block to prepare error notifications for base configurator.